### PR TITLE
fix(router): Fix model management when going "back" through history.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
#### What is the problem?
When going "back" through history, the router creates a new model
for the previous view. This causes problems in email-first where
the user's email address on /signup_password is filled in from
an `account` model. Users that go to the TOS/PP agreements then
click "back" are sent all the way back to `/email` because a new
view is created for /signup_password.

#### How does this fix the problem?
Keep a stack of viewModels. When going back through history,
re-use the same model that was used the first time a view is
displayed.

extraction from #5177

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

File: app/scripts/lib/router.js [hunk 1]
```diff
@@ -138,17 +139,30 @@ define(function (require, exports, module) {
...
+     * @param {String} url
+     * @param {Object} [nextViewData={}]
+     * @param {Object} [options={}]
+     *   @param {Boolean} [options.clearQueryParams=false] Clear the query parameters?
+     *   @param {Boolean} [options.replace=false] Replace the current view?
+     *   @param {Boolean} [options.trigger=true] Show the new view?
+     * @returns {any}
+     */
+    navigate (url, nextViewData = {}, options = {}) {
+      if (options.replace && this._viewModelStack.length) {
```
Comment (shane-tomlinson) [inline app/scripts/lib/router.js]: This branch needs a test

Comment (shane-tomlinson) [inline app/scripts/lib/router.js]: added

**Comment by philbooth** [comment]
LGTM, r+

**Pull request merged**